### ISSUE TITLE
Bad error reporting for circular imports

### ISSUE THREAD

**Issue by ysndr** [tweag/nickel#407]
**Describe the bug**
In case of circular imports the resulting error is:

```
error: Internal error (Resolved import not found (FileId(1)))
  ┌─ /Volumes/projects/nickel/examples/b.ncl:1:9
  │
1 │ let a = import "a.ncl" in a
  │         ^^^^^^^^^^^^^^ here
  │
  = This
```

**To Reproduce**
Create two files:

`a.ncl`
```
let b = import "b.ncl" in b
```

`b.ncl`
```
let a = import "a.ncl" in a
```

run nickel on a or b (`nickel  -f a.ncl `)

**Expected behavior**
A message hinting on a circular import (is there any other reason why there could be an existing FileId for the import ed file?)

**Environment**
 - MacOS
 - Version of the code: c15e183c03ebdd4a3f46dc3373e60b76a425d0b1

**Comment by yannham** [comment]
In that case, i think the problem is that `a.ncl` is the main file, so it's not added to the import resolver. It shouldn't be that hard to fix. In Nickel, cyclic imports aren't an error, they just introduce potentially mutually recursive computations.

However, what's more concerning is that if you add a `c.ncl` with `let y = import "a.ncl" in 0`, then typechecking or running it overflows the stack. It didn't use to be the case. I'll open a separate issue.

**Comment by francois-caddet** [comment]
fixed by #432 
Do not produce anymore unexpected error neither stack overflow.
Now it loop indefinitly as expected.
If the infinit loop is an issue. open a new one.
